netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java

```diff
@@ -190,7 +242,42 @@ public void onReady() {
     }
   }
 
+  private static final class TestServerStreamListener implements ServerStreamListener {
+    final ServerStream stream;
+    final String method;
+    final Metadata.Headers headers;
+
+    TestServerStreamListener(ServerStream stream, String method, Metadata.Headers headers) {
+      this.stream = stream;
+      this.method = method;
+      this.headers = headers;
+      stream.request(1);
+    }
+
+    @Override
+    public void messageRead(InputStream message) {
+      // Just echo back the message.
+      stream.writeMessage(messageStream());
```

Comment:
To truly echo back, we would need to do `stream.writeMessage(message)`. I see that this was just moved, and it was incorrect before.